File: cpp-code/List.cpp
//
// Created by danis on 9/6/2023.
//
#include <iostream>
#include "List.h"
#include "Node.h"
#include "Data.h"

List::List() {
    this->head = nullptr;
    this->tail = nullptr;
    this->event = "A";
}

void List::add() {
    Node *node = new Node(new Data());

    if (this->head == nullptr) {
        this->head = node;
        this->tail = node;
    } else {
        this->tail->setNext(node);
        node->setPrev(this->tail);
        this->tail = node;
    }
}

void List::showList() {
    if (this->head == nullptr) {
        std::cout << "List is empty" << std::endl;
        return;
    }
    Node *node = this->head;
    while (node != nullptr) {
        std::cout << node->toString();
        node = node->getNext();
    }
}

void List::searchForEvent() {
    if (this->tail != nullptr) {
        if (this->tail->getData()->isPrime()) {
            // Here will come the confirmation for every event that could happen
            this->tail->getData()->setEvent(this->event);
        }
    }
}

File: python-code/list.py
from data import Data
from node import Node

class List():
    def __init__(self):
        self.head = None
        self.tail = None

    def append(self):
        node = Node(Data())
        if self.head is None:
            self.head = node
            self.tail = node
        else:
            self.tail.next = node
            node.prev = self.tail
            self.tail = node

    def showList(self):
        if self.head is None:
            print('Empty list')
            return
        node = self.head
        while node:
            print(node, end=' ')
            node = node.next